bundle/deploy/terraform/plan.go

```diff
@@ -9,8 +9,54 @@ import (
 	"github.com/databricks/cli/libs/cmdio"
 	"github.com/databricks/cli/libs/terraform"
 	"github.com/hashicorp/terraform-exec/tfexec"
+	tfjson "github.com/hashicorp/terraform-json"
 )
 
+// printPlanSummary prints a high level summary of the terraform plan that will
+// be applied during bundle deploy / destroy.
+func printPlanSummary(ctx context.Context, plan *tfjson.Plan) error {
+	cmdio.LogString(ctx, "\nPlan:")
+	for _, change := range plan.ResourceChanges {
+		tfActions := change.Change.Actions
+		if tfActions.Read() || tfActions.NoOp() {
+			continue
+		}
+
+		var action string
+		switch {
+		case tfActions.Update():
+			action = "Update"
+		case tfActions.Create():
+			action = "Create"
+		case tfActions.Delete():
+			action = "Delete"
+		case tfActions.Replace():
+			action = "Replace"
```

Comment:
Could this be called "Update"? Which better explains this concept for users (we don't actually maintain state!)

Reply:
`Replace` has side effects that update does not. This is worth highlighting for users. For example, if a DLT pipeline is "replaced" or "recreated" then associated metadata with past runs is lost. This means all tables would have to be recomputed.

I am open to changing what we call this though, would "recreate" capture this better?